pkg/sdkserver/sdkserver_test.go

```diff
@@ -1401,9 +1403,9 @@ func TestSDKServerGracefulTerminationInterrupt(t *testing.T) {
 	//	Assert gs is still requestReady
 	assert.Equal(t, agonesv1.GameServerStateRequestReady, sc.gsState)
 	// gs Shutdown
-	_, err = sc.Shutdown(sdkCtx, &sdk.Empty{})
-	require.NoError(t, err)
-	assert.Equal(t, agonesv1.GameServerStateShutdown, sc.gsState)
+	gs.Status.State = agonesv1.GameServerStateShutdown
+	fakeWatch.Modify(gs.DeepCopy())
```

Comment:
It looks like all of the tests in this file now use `fakeWatch.Modify()` instead of calling `sc.Shutdown()`. Should we have unit tests for both?

Most of the lines to the shutdown state in https://agones.dev/site/docs/reference/gameserver/#gameserver-state-diagram are through the SDK.

Reply:
> It looks like all of the tests in this file now use `fakeWatch.Modify()` instead of calling `sc.Shutdown()`. Should we have unit tests for both?

https://github.com/googleforgames/agones/blob/7492d86c2f55b3980050522f961f886507abbd96/pkg/sdkserver/sdkserver_test.go#L43

Has all the unit tests to ensure the SDK moves the GameServer to the correct state. We also have e2e tests to confirm that `sdk.Shutdown()` does move the `GameServer` to shutdown as well.

https://github.com/googleforgames/agones/blob/7492d86c2f55b3980050522f961f886507abbd96/test/e2e/gameserver_test.go#L693

So I don't think we need a separate `sdk.Shutdown()` unit test for this use case, since we already have tests showing that it will successfully move the `GameServer` to a `Shutdown` state.

This test confirms that no matter how a `GameServer` gets to a `Shutdown` state, it responds as expected.

Reply:
Makes sense.